hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java

```diff
@@ -44,8 +50,20 @@ public class ContractCreateHandler implements TransactionHandler {
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txBody,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup) {
+        final var op = txBody.getContractCreateInstance();
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txBody).payerKeyFor(payer);
+        final var adminKey = asHederaKey(op.getAdminKey());
+        if (adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
+            meta.addToReqNonPayerKeys(adminKey.get());
+        }
+        if (op.hasAutoRenewAccountId() && !op.getAutoRenewAccountId().equals(MISSING_ACCOUNT_ID)) {
```

Comment:
Imo we can leave off this `!op.getAutoRenewAccountId().equals(MISSING_ACCOUNT_ID)` test; it's just a small optimization to save calling `addNonPayerKey()` when it's guaranteed to fail; but seems very arbitrary. 

(I.e. why don't we also check if `accountNum < 0`, or `shardNum != STATIC_PROPERTIES_HOLDER.shard()`, etc., etc...?)

Reply:
Agreed. Removed that check in `ContractCreate` preHandle